src/pages/Home.js

```diff
@@ -1,85 +1,115 @@
-    // },
-    // {
-    //   id: 6,
-    //   title: "Natural, The",
-    //   description: "Drama",
-    //   year: 1973,
-    //   image:
-    //     "https://images.unsplash.com/photo-1536440136628-849c177e76a1?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8NHx8bW92aWV8ZW58MHx8MHx8&auto=format&fit=crop&w=900&q=60",
-    // },
-  ];
+  const [display, setDisplay] = useState(false);
+  // const [movies, setMovies] = useState([]);
+  // const [searchTerm, setSearchTerm] = useState("");
+
+  const FilterIdSelector = useSelector((state) => state.id);
+  const movieGenreFilterSelector = useSelector((state) => state.filterMovie);
+  const apiMoviesArraySelector = useSelector((state) => state.movies);
+  const movieArraySearchSelector = useSelector((state) => state.searchFilter);
+
+  // const addMovieHandler = (movie) => {
+  //   setMovies((prevMovies) => {
+  //     return [movie, ...prevMovies];
+  //   });
+  // };
+
+  const handleDisplay = () => {
+    setDisplay(!display);
+  };
+
   return (
     <>
-      <div className="banner">
-        <div className="inner-banner">
-          <h1>FIND YOUR MOVIE</h1>
-          <form>
-            <input type="text" placeholder="What do you want to watch?" />
-            <button type="submit">search</button>
-          </form>
-        </div>
+      <div className={FilterIdSelector.title !== "" ? "hide" : "banner"}>
+        <InnerBanner
+          // searchTerm={searchTerm}
+          // onAddMovie={addMovieHandler}
+          close={handleDisplay}
+        />
       </div>
-      <div className="card-layout">
-        <ErrorBoundary>
-          {movieData.map((movie) => (
-            <MovieCard
-              key={movie.id}
-              image={movie.image}
-              title={movie.title}
-              description={movie.description}
-              year={movie.year}
-            />
-          ))}
-        </ErrorBoundary>
+
+      <div className={FilterIdSelector.title !== "" ? "movie-details" : null}>
+        <MovieBanner
+          title={FilterIdSelector.title}
+          vote_average={FilterIdSelector.vote_average}
+          poster_path={FilterIdSelector.poster_path}
+          tagline={FilterIdSelector.tagline}
+          genres={FilterIdSelector.genres}
+          release_date={FilterIdSelector.release_date.slice(0, 4)}
+          runtime={FilterIdSelector.runtime}
+          overview={FilterIdSelector.overview}
+          id={FilterIdSelector.id}
+          broken={(e) =>
+            (e.target.src =
+              "https://images.unsplash.com/photo-1560109947-543149eceb16?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1035&q=80")
+          }
+        />
       </div>
+      <Navigation
+        all=""
+        action="Action"
+        fantasy="Fantasy"
+        drama="Drama"
+        crime="Crime"
+      />
+      <ErrorBoundary>
+        <div className="card-layout">
+          {apiMoviesArraySelector.map((movie) =>
+            movie
```

Comment:
you're doing search logic yourself, on the frontend which is not an ideal solution. It should be a request to the API that'll return you data of the movies you're looking for